twarep/electrifly
# Upload screen recent records table

## Checklist
**Note:** Put an x into the box like this [x] to check these off

## What I Did
- Created the flight_weather_data_view in the database 

<img width="1037" alt="Screen Shot 2023-07-25 at 2 16 36 AM" src="https://github.com/twarep/electrifly/assets/71115362/20a1a985-bdcf-4735-a280-3e0075be7a56">

- Upload screen most recent records table 

![Screen Shot 2023-07-25 at 2 18 02 AM](https://github.com/twarep/electrifly/assets/71115362/1baed71b-4683-4bd9-b8c4-1c667c208ec9)

## How I Did it?
All required imports should be included in app.py, but I used psycopg2 and sqlalchemy to do this in the UI, and SQL query script for the view in the database. 

## How to Test?
Manual test, make sure the table populates, make sure all the columns are there. 
Check [This Notion Page](https://www.notion.so/UI-Upload-Screen-Most-Recent-Records-Table-7b781bd1542f48bd86c0987789460c4b) to ensure all the appropriate columns are shown in the collapsed mode. 

### Relevant Links
Any tutorials or other resources used to help with this task. You can also cite the documentation.
- https://appsilon.com/shiny-for-python-databases-mysql-postgres/#:~:text=Shiny%20for%20Python%20%E2%80%93%20How%20to%20Connect%20to%20Databases,-The%20dashboard%20you&text=You%20don't%20have%20to,to%20the%20database%20%E2%80%93%20connect_to_db()%20
- https://shiny.posit.co/py/api/reactive.Effect.html
